code/game/machinery/dna_infuser/organ_sets/fly_organs.dm

```diff
@@ -0,0 +1,151 @@
+	. = ..()
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/new_fly = owner
+	if(isflyperson(new_fly))
+		return
+	//okay you NEED to be a fly
+	to_chat(new_fly, span_danger("Too much fly DNA! Your skin begins to discolor into a horrible black as you become more fly than person!"))
+	new_fly.set_species(/datum/species/fly)
+
+/datum/status_effect/organ_set_bonus/fly/disable_bonus()
+	. = ..()
+	//sorry dude
+
+/obj/item/organ/internal/eyes/fly
+	name = "fly eyes"
+	desc = "These eyes seem to stare back no matter the direction you look at it from."
+	eye_icon_state = "flyeyes"
+	icon_state = "eyeballs-fly"
+	flash_protect = FLASH_PROTECTION_HYPER_SENSITIVE
+	native_fov = NONE //flies can see all around themselves.
+
+/obj/item/organ/internal/eyes/fly/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/organ_set_bonus, /datum/status_effect/organ_set_bonus/fly)
+
+/obj/item/organ/internal/tongue/fly
+	name = "proboscis"
+	desc = "A freakish looking meat tube that apparently can take in liquids."
+	icon = 'icons/obj/medical/organs/fly_organs.dmi'
+	say_mod = "buzzes"
+	taste_sensitivity = 25 // you eat vomit, this is a mercy
+	modifies_speech = TRUE
+	languages_native = list(/datum/language/buzzwords)
+	var/static/list/languages_possible_fly = typecacheof(list(
+		/datum/language/common,
+		/datum/language/draconic,
+		/datum/language/codespeak,
+		/datum/language/monkey,
+		/datum/language/narsie,
+		/datum/language/beachbum,
+		/datum/language/aphasia,
+		/datum/language/piratespeak,
+		/datum/language/moffic,
+		/datum/language/sylvan,
+		/datum/language/shadowtongue,
+		/datum/language/terrum,
+		/datum/language/nekomimetic,
+		/datum/language/buzzwords
+	))
+
+/obj/item/organ/internal/tongue/fly/modify_speech(datum/source, list/speech_args)
+	var/static/regex/fly_buzz = new("z+", "g")
+	var/static/regex/fly_buZZ = new("Z+", "g")
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = fly_buzz.Replace(message, "zzz")
+		message = fly_buZZ.Replace(message, "ZZZ")
+		message = replacetext(message, "s", "z")
+		message = replacetext(message, "S", "Z")
+	speech_args[SPEECH_MESSAGE] = message
+
+/obj/item/organ/internal/tongue/fly/Initialize(mapload)
+	. = ..()
+	languages_possible = languages_possible_fly
+	AddElement(/datum/element/organ_set_bonus, /datum/status_effect/organ_set_bonus/fly)
+
+/obj/item/organ/internal/heart/fly
+	desc = "You have no idea what the hell this is, or how it manages to keep something alive in any capacity."
+
+/obj/item/organ/internal/heart/fly/Initialize(mapload)
+	. = ..()
+	name = odd_organ_name()
+	icon_state = pick("brain-x-d", "liver-x", "kidneys-x", "spinner-x", "lungs-x", "random_fly_1", "random_fly_2", "random_fly_3", "random_fly_4", "random_fly_5")
+	AddElement(/datum/element/organ_set_bonus, /datum/status_effect/organ_set_bonus/fly)
+
+/obj/item/organ/internal/heart/fly/update_icon_state()
+	SHOULD_CALL_PARENT(FALSE)
```

Comment:
dont you want update icon blocker in that situation?

Reply:
that's not code i added!